src/Lowranksvd.jl

```diff
@@ -1,5 +1,163 @@
 module Lowranksvd
+# Low Rank Singular Value Decomposition
+using LinearAlgebra
+import Base.show
+"""
+    LowRankSVD <: Factorization
+Matrix factorization type of the singular value decomposition (SVD) approximation of a low rank matrix `A`.
+LowRankSVD is return type of [`svd_lowrank(_)`](@ref), the corresponding matrix factorization function.
+If `F::SVD` is the factorization object, `U`, `S`, `V` and `Vt` can be obtained
+via `F.U`, `F.S`, `F.V` and `F.Vt`, such that `A ≈ U * Diagonal(S) * Vt`.
+The singular values in `S` are sorted in descending order.
+"""
+struct LowRankSVD{T,Tr,M<:AbstractArray{T}} <: Factorization{T}
+    U::M
+    S::Vector{Tr}
+    Vt::M
+    function LowRankSVD{T,Tr,M}(U, S, Vt) where {T,Tr,M<:AbstractArray{T}}
+      size(U,2) == size(Vt,1) == length(S) || throw(DimensionMismatch("$(size(U)), $(length(S)), $(size(Vt)) not compatible"))
+      new{T,Tr,M}(U, S, Vt)
+    end
+end
+LowRankSVD(U::AbstractArray{T}, S::Vector{Tr}, Vt::AbstractArray{T}) where {T,Tr} = LowRankSVD{T,Tr,typeof(U)}(U, S, Vt)
+function LowRankSVD{T}(U::AbstractArray, S::AbstractVector{Tr}, Vt::AbstractArray) where {T,Tr}
+    LowRankSVD(convert(AbstractArray{T}, U),
+                convert(Vector{Tr}, S),
+                convert(AbstractArray{T}, Vt))
+end
+function Base.show(io::IO, mime::MIME{Symbol("text/plain")}, F::LowRankSVD{<:Any,<:Any,<:AbstractArray})
+    summary(io, F); println(io)
+    println(io, "U factor:")
+    show(io, mime, F.U)
+    println(io, "\nsingular values:")
+    show(io, mime, F.S)
+    println(io, "\nVt factor:")
+    show(io, mime, F.Vt)
+end
+"""
+    get_approximate_basis(A, l::Int64; niter::Int64 = 2, M = nothing) -> Q
+
+Return Matrix ``Q``` with ``l`` orthonormal columns such that ``Q Q^H A`` approximates ``A``. If ``M`` is specified, then ``Q`` is such that ``Q Q^H (A - M)`` approximates ``A - M``.
+    
+# Arguments
+- `A::AbstractArray{T}`: the input matrix of size ``(m, n)``.
+- `l::Int64`: the dimension of subspace spanned by Q columns.
+- `niter::Int64`(optional): the number of subspace iterations to conduct; `niter` must be a nonnegative integer. In most cases, the default value 2 is more than enough.
+- `M::AbstractArray{T}`(optional): the input matrix of size ``(m, n)``.
+
+# Examples
+```jloctest
+julia> A = rand(3,3);Q = get_approximate_basis(A,2)
+3×2 Matrix{Float64}:
+ -0.737784  -0.199989
+ -0.451893  -0.563196
+ -0.501465   0.801757
+```
+
+# References
+- Nathan Halko, Per-Gunnar Martinsson, and Joel Tropp, Finding structure with randomness: probabilistic algorithms for constructing approximate matrix decompositions, arXiv:0909.4061 [math.NA; math.PR], 2009 (available at `arXiv <http://arxiv.org/abs/0909.4061>`_).
+"""
+function get_approximate_basis(
+    A::AbstractArray{T}, l::Int64, niter::Int64 = 2, M::Union{AbstractArray{T}, Nothing} = nothing) where T
```

Comment:
> https://docs.julialang.org/en/v1/manual/integers-and-floating-point-numbers/#Integers
> Julia also defines the types Int and UInt, which are aliases for the system's signed and unsigned native integer types respectively

在 64 位系统上 `Int === Int64`, 在 32 位系统上 `Int === Int32`.

```suggestion
    A::AbstractArray{T}, l::Int, niter::Int = 2, M::Union{AbstractArray{T}, Nothing} = nothing) where T
```

相关的地方都修改成 `Int` 之后CI应该就能通过了。

Reply:
好的，感谢老师~在[生成测试用例](https://github.com/jieli-matrix/Lowranksvd.jl/blob/dev/test/lowranksvd.jl#L5)也需要考虑对应的问题，实数没问题，但复数似乎要考虑rand(ComplexF32,(n,m))的问题（我试了下Complex是无法根据系统位数生成对应矩阵）。我能想到的是实部与虚部分别生成，这样应该能够全平台测试，如果老师有更好的建议，也请告诉我~